task-74

marsel-code opened the pull request:
Обработал нажатие на кнопку share на экране Vacancy.

Commit:
task-74
Обработал нажатие на кнопку share на экране Vacancy.

Fargo02 commented on app/src/main/java/ru/practicum/android/diploma/data/ExternalNavigatorImpl.kt:
```diff
@@ -0,0 +1,29 @@
+        shareIntent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK)
+        shareIntent.setType("text/playn")
+        shareIntent.putExtra(Intent.EXTRA_TEXT, url)
+        context.startActivity(shareIntent)
```
Стоит добавить try-catch в случае если на телефоне не будет ни одного приложение через которое можно поделиться и вывести например toast "Нет подходящего приложения"

Fargo02 commented on app/src/main/java/ru/practicum/android/diploma/ui/vacancy/VacancyViewModel.kt:
```diff
@@ -22,11 +23,13 @@ class VacancyViewModel(
 
     private val _state = MutableLiveData<VacancyDetailsState>()
     val state: LiveData<VacancyDetailsState> get() = _state
+    private var vacancyUrl: String = ""
```
Можно заменить на константу, не зря же мы ее придумали)
 private var vacancyUrl: String = EMPTY_PARAM_VALUE

Fargo02 commented on app/src/main/java/ru/practicum/android/diploma/ui/vacancy/VacancyViewModel.kt:
```diff
@@ -38,7 +41,7 @@ class VacancyViewModel(
             else -> Payload(details.first!!)
         }
-
+        if (details.first?.url != "") vacancyUrl = details.first!!.url
```
Тут тоже константу бахни

Commit:
task-74
Обработал нажатие на кнопку share на экране Vacancy. добавид обработку исключений

pyclan88 commented on app/src/main/java/ru/practicum/android/diploma/data/ExternalNavigatorImpl.kt:
```diff
@@ -0,0 +1,36 @@
+            shareIntent.setType("text/playn")
+            shareIntent.putExtra(Intent.EXTRA_TEXT, url)
+            context.startActivity(shareIntent)
+        } catch (ex: Exception) {
```
подавлением лечится
@Suppress("TooGenericExceptionCaught")

pyclan88 commented on app/src/main/java/ru/practicum/android/diploma/data/ExternalNavigatorImpl.kt:
```diff
@@ -0,0 +1,36 @@
+            shareIntent.setType("text/playn")
+            shareIntent.putExtra(Intent.EXTRA_TEXT, url)
+            context.startActivity(shareIntent)
+        } catch (ex: Exception) {
```
аналогично
@Suppress("SwallowedException")

Commit:
task-74
Обработал нажатие на кнопку share на экране Vacancy. исправил замечания CI

Commit:
task-74
Обработал нажатие на кнопку share на экране Vacancy. исправил замечания CI

Commit:
task-74
Обработал нажатие на кнопку share на экране Vacancy. исправил замечания CI
